crates/service/src/algorithms/quantization/scalar.rs

```diff
@@ -40,18 +40,23 @@ impl<S: G> ScalarQuantization<S> {
 
 impl<S: G> Quan<S> for ScalarQuantization<S> {
     fn create(
-        path: PathBuf,
+        path: &Path,
         options: IndexOptions,
         _: QuantizationOptions,
-        raw: &Arc<Raw<S>>,
+        raw: &Arc<Raw<S::Storage>>,
     ) -> Self {
-        std::fs::create_dir(&path).unwrap();
+        debug_assert!(
```

Comment:
Where is feature gate? I didn't see it in a `impl Validate`.

Reply:
Previously, it was checked in `crates/service/src/instance/mod.rs` manually. Now, it tells `validator_derive` to use `validate_index_options` function.